src/zenml/new/pipelines/pipeline.py

```diff
@@ -1390,11 +1390,59 @@ def _parse_config_file(
                         "model_config"
                     ]
                 else:
-                    from zenml.model.model_config import ModelConfig
+                    from zenml.model.model_config import (
+                        ModelVersionConsumerConfig,
+                        ModelVersionProducerConfig,
+                    )
 
-                    _from_config_file["model_config"] = ModelConfig.parse_obj(
-                        _from_config_file["model_config"]
+                    producer_init_args = (
+                        ModelVersionProducerConfig.__init__.__code__.co_varnames
+                    )
+                    consumer_init_args = (
+                        ModelVersionConsumerConfig.__init__.__code__.co_varnames
                     )
+                    producer_exclusive_init_args = set(
+                        producer_init_args
+                    ) - set(consumer_init_args)
+                    if any(
+                        [
+                            k in producer_exclusive_init_args
+                            for k in _from_config_file["model_config"].keys()
+                        ]
+                    ):
+                        logger.info(
+                            "`model_config` identified as producer configuration, "
+                            "if this is not correct consider using only following "
+                            f"arguments in you configuration file: {consumer_init_args}."
+                        )
+                        _from_config_file[
+                            "model_config"
+                        ] = ModelVersionProducerConfig(
+                            **{
+                                k: v
+                                for k, v in _from_config_file[
+                                    "model_config"
+                                ].items()
+                                if k in producer_init_args
+                            }
+                        )
+                    else:
+                        logger.info(
+                            "`model_config` identified as consumer configuration, "
+                            "if this is not correct consider setting one of following "
+                            f"arguments in you configuration file: {producer_exclusive_init_args}."
+                        )
+                        _from_config_file[
+                            "model_config"
+                        ] = ModelVersionConsumerConfig(
+                            **{
+                                k: v
+                                for k, v in _from_config_file[
+                                    "model_config"
+                                ].items()
+                                if k in consumer_init_args
+                            }
+                        )
```

Comment:
I don't like this part much, cause it is rather implicit and may behave awkwardly sometimes.

Example of consumer config:
```yaml
model_config:
    name: "foo"
```
Example of producer config:
```yaml
model_config:
    name: "foo"
    license: "apache"
```

Any thoughts on how to make it better? Introducing two separate options (e.g. "model_consumer_config") or adding extra field (say "type":"producer")?